File: reto15/reto15.ts
function drawTable(data: Array<Record<string, string | number>>): string {
  if (!data.length) return "+\n+";

  const keys = Object.keys(data[0]);
  const lengths: Record<string, number> = {};

  for (const key of keys) {
    let maxLength = key.length;
    for (const row of data) {
      const len = String(row[key]).length;
      if (len > maxLength) maxLength = len;
    }
    lengths[key] = maxLength;
  }

  let separator = "+";
  for (const key of keys) {
    separator += `${"-".repeat(lengths[key] + 2)}+`;
  }

  let header = "|";
  for (const key of keys) {
    header += `${` ${key[0].toUpperCase()}${key.slice(1)}`.padEnd(
      lengths[key] + 2
    )}|`;
  }

  const rows: string[] = [];
  for (const row of data) {
    let line = "|";
    for (const key of keys) {
      line += `${` ${String(row[key])}`.padEnd(lengths[key] + 2)}|`;
    }
    rows.push(line);
  }

  return `${separator}\n${header}\n${separator}\n${rows.join(
    "\n"
  )}\n${separator}`;
}

File: reto15/reto15.py
def drawTable(data):
    if not data:
        return "+\n+"

    keys = list(data[0].keys())
    lengths = {}

    for key in keys:
        max_length = len(key)
        for row in data:
            length = len(str(row[key]))
            if length > max_length:
                max_length = length
        lengths[key] = max_length

    separator = "+"
    for key in keys:
        separator += "-" * (lengths[key] + 2) + "+"

    header = "|"
    for key in keys:
        header += f" {key[0].upper()}{key[1:]}".ljust(lengths[key] + 2) + "|"

    rows = []
    for row in data:
        line = "|"
        for key in keys:
            line += f" {str(row[key])}".ljust(lengths[key] + 2) + "|"
        rows.append(line)

    return f"{separator}\n{header}\n{separator}\n{chr(10).join(rows)}\n{separator}"
